src/screens/Exchange/Exchange.js

```diff
@@ -242,106 +224,61 @@ class ExchangeScreen extends React.Component<Props, State> {
-      }
-      : {
-        fromAmount: val,
-        fromAmountInFiat: String(getBalanceInFiat(baseFiatCurrency, val, rates, symbol)),
-      },
-    );
-    this.setErrorMessage(getErrorMessage(val, fromAsset));
-  };
-
   getFromInput = () => {
-    const { baseFiatCurrency } = this.props;
     const {
-      errorMessage, fromAsset, fromAmount, fromAmountInFiat, displayFiatFromAmount,
+      fromAsset, fromAmount,
     } = this.state;
-    const { assetBalance, symbol = '' } = fromAsset;
-    const value = displayFiatFromAmount ? fromAmountInFiat : fromAmount;
+    const { fromOptions, horizontalOptions } = this.options;
 
     return (
-      <TextInput
+      <ValueInput
+        assetData={fromAsset}
+        onAssetDataChange={(assetData) => this.setState({ fromAsset: assetData })}
+        value={fromAmount}
+        onValueChange={amount => this.setState({ fromAmount: amount })}
+        selectorOptionsTitle={t('label.sell')}
+        customAssets={fromOptions}
+        horizontalOptions={horizontalOptions}
+        leftSideSymbol="-"
         getInputRef={ref => { this.fromInputRef = ref; }}
-        onChange={this.handleFromInputChange}
-        value={value}
-        onFocus={this.onFocusInput}
         onBlur={this.blurFromInput}
-        errorMessage={errorMessage}
-        asset={fromAsset}
-        onAssetPress={() => this.setState({ showSellOptions: true })}
-        labelText={assetBalance && getFormattedSellMax(fromAsset)}
-        onLabelPress={() => this.handleUsePercent(100)}
-        leftSideText={displayFiatFromAmount
-          ? t('tokenValue', {
-            value: formatAmount(fromAmount || '0', 2) || '0',
-            token: fromAsset.symbol || '',
-          })
-          : formatFiat(fromAmountInFiat, baseFiatCurrency).replace(/ /g, '')
-        }
-        leftSideSymbol="-"
-        onLeftSideTextPress={() => this.setState({ displayFiatFromAmount: !displayFiatFromAmount })}
-        rightPlaceholder={displayFiatFromAmount ? baseFiatCurrency || defaultFiatCurrency : symbol}
-        inputAccessoryViewID={INPUT_ACCESSORY_NATIVE_ID}
+        onFormValid={valid => this.setState({ isFormValid: valid })}
       />
     );
   };
 
   getToInput = () => {
-    const { baseFiatCurrency, offers, rates } = this.props;
+    const { offers } = this.props;
     const {
-      displayFiatToAmount, toAsset, fromAmount,
+      toAsset, fromAmount,
     } = this.state;
+    const { toOptions, horizontalOptions } = this.options;
 
-    let value;
-    let toAmount;
-    let toAmountInFiat;
+    let toAmount = '0';
     if (offers?.length && fromAmount) {
-      toAmount = getBestAmountToBuy(offers, fromAmount);
-      if (toAmount) {
-        toAmountInFiat = formatAmount(getBalanceInFiat(baseFiatCurrency, toAmount, rates, toAsset.symbol || ''), 2);
-        value = displayFiatToAmount ? toAmountInFiat : formatAmount(toAmount || '', 6);
-      }
+      toAmount = getBestAmountToBuy(offers, fromAmount) || '0';
```

Comment:
We used to shorten this value. 
Now if value has multiple decimal points, it's hard to comprehend it. Especially when input is disabled and there's no way to "scroll" the value.